_posts/2020-08-11-pull-requests-require-context.md

```diff
@@ -0,0 +1,160 @@
+next change. It's like he's sitting next to you!
+
+## Show Your Work
+
+If your PR contains work that is beyond trivial, show your reviewers how you thought about the problem. Demonstrate
+the effects of the changes. Give them confidence that you've worked through this problem thoroughly, and you've
+brought receipts.
+
+### Make small, self-contained commits
+
+A good PR starts with good commits. Good commits are small, self-contained, and leave the codebase always in a
+working state. With good commits, reviewers can see exactly how you worked through the problem you were solving.
+[Here's a PR](https://github.com/artsy/convection/pull/645) from [Jon](https://github.com/jonallured) that
+demonstrates the use of small, self-contained commits to describe his approach to refactoring code before fixing a
+bug.
+
+_Bonus tip_: it can be easier to review PRs with many small commits via the
+[_Commits_](https://github.com/artsy/convection/pull/645/commits) tab instead of the
+[_Files changed_](https://github.com/artsy/convection/pull/645/files) tab.
+
+### Demonstrate the results
+
+Pictures are a worth a thousand words. Animated gifs are worth a thousand pictures (uhhhh, in file size too 😬). An
+animated gif showing the outcome of your PR gives reviewers a demo, and confidence that you've verified your
+changes.
+
+[Here's a PR](https://github.com/artsy/force/pull/5817) from [Ashley](https://github.com/ashleyjelks) that includes
+animated gifs of the changes she's made. The effects of the changes might not be obvious by looking only at the
+code, but seeing them in action makes it clear.
+
+### Document the unseen
+
+Sometimes a PR's changes have effects outside of the UI. There are still ways to give reviewers proof that the
+changes have the desired effects.
+
+[Here's a PR](https://github.com/artsy/eigen/pull/3206) from [Yuki](https://github.com/yuki24) that not only
+demonstrates what's happening in the UI, but also assures me that the back-end data is getting updated properly
+through a Rails console.
+
+[Christina](https://github.com/xtina-starr) authored [this PR](https://github.com/artsy/reaction/pull/3441) which
+shows the UI changes in addition to some output from her browser console, demonstrating that analytics tracking
+calls are firing correctly.
+
+### Share your progress
+
+One mistake many engineers make with non-trivial pull requests is to wait to open them until they're "done". If
+there are changes you'd like to get people's eyes on quickly, open a WIP PR before the work is done: mark it as a
+draft in GitHub, or put `WIP` in the title. Extra work up front avoids rework by starting early discussions about
+your approach.
+
+Let reviewers know in the body that your work isn't complete. As you continue your work, use a `TODO:` list in the
+body to illustrate your progress, as in [this PR](https://github.com/artsy/palette/pull/464) from
+[Sepand](https://github.com/sepans).
```

Comment:
@sepans any objections to me referencing you & your PR here?

Reply:
Not at all that's an honor 😄